### PR TITLE
fix(e2e): ignore error log about failed storage update

### PR DESCRIPTION
**Pull request by ghouscht**
Should fix https://github.com/etcd-io/etcd/issues/18978 as suggested by @ahrtr

### CODE REVIEW

File: tests/e2e/logging_test.go [hunk 1]
```diff
@@ -37,15 +37,22 @@ func TestNoErrorLogsDuringNormalOperations(t *testing.T) {
 				e2e.WithClusterSize(1),
 				e2e.WithLogLevel("debug"),
 			},
-			allowedErrors: map[string]bool{"setting up serving from embedded etcd failed.": true},
+			allowedErrors: map[string]bool{
+				"setting up serving from embedded etcd failed.": true,
+			},
```
Comment (ahrtr) [inline tests/e2e/logging_test.go:42]: Why don't we ignore the error in such case? Suggest to add it into this case as well.

File: tests/e2e/logging_test.go [hunk 2]
```diff
@@ -37,15 +37,22 @@ func TestNoErrorLogsDuringNormalOperations(t *testing.T) {
...
 			options: []e2e.EPClusterOption{
 				e2e.WithClusterSize(3),
 				e2e.WithLogLevel("debug"),
 			},
-			allowedErrors: map[string]bool{"setting up serving from embedded etcd failed.": true},
+			allowedErrors: map[string]bool{
+				"setting up serving from embedded etcd failed.": true,
+				// See https://github.com/etcd-io/etcd/pull/19040#issuecomment-2539173800
+				// TODO: Remove with etcd 3.7
+				"failed to update storage version": true,
```
[the remark this answers is not in the document]
Comment (ghouscht) [reply tests/e2e/logging_test.go:54]: The test seems to check the `msg` not `error` object in the logged JSON: https://github.com/ghouscht/etcd/blob/ignore-missing-err-confstate/tests/e2e/logging_test.go#L136-L139

File: tests/e2e/logging_test.go (hunk 1 above)
Comment (ghouscht) [reply tests/e2e/logging_test.go:42]: We can ignore it here as well, but according to testgrid there was no failure for this test. Only the three node cases were affected from what I can see.

File: tests/e2e/logging_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahrtr) [reply tests/e2e/logging_test.go:54]: I think we need to check both the msg and error. In this case, it makes more sense to ignore the specific error message `cannot detect storage schema version: missing confstate information`.

We can add one more field "`Error`" into `logEntry`.

```
type logEntry struct {
	Level     string `json:"level"`
	Timestamp string `json:"ts"`
	Caller    string `json:"caller"`
	Message   string `json:"msg"`
}
```

File: tests/e2e/logging_test.go (hunk 1 above)
Comment (ahrtr) [reply tests/e2e/logging_test.go:42]: In theory, one member may still run into the issue, but just with a low possibility.

Comment (ghouscht) [reply tests/e2e/logging_test.go]: Ok, I added the exclusion for this test as well.

File: tests/e2e/logging_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghouscht) [reply tests/e2e/logging_test.go]: Adapted the code to check the Message and Error for a match. If one of the two matches exactly the error is ignored. Hope that is ok

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you!

**Comment by ahrtr** [comment]
cc @ivanvc @jmhbnz @serathius

**Review by ivanvc — APPROVED** [review]
I'm not an expert on this, but it LGTM. Anything that helps with flakes is greatly appreciated. Thanks, @ghouscht.

**Review by jmhbnz — APPROVED** [review]
LGTM - Many thanks for working on this @ghouscht 🙏🏻

**Pull request merged**